src/puj/push/instructions/numeric.clj

```diff
@@ -1,31 +1,28 @@
 (ns puj.push.instructions.numeric
-  (:require [puj.push.unit :as u]))
+  (:require [puj.push.instruction :as i]))
 
 
 (defn instructions []
-  {
-   :int-add
+  [
    (with-meta
-     (u/simple-instruction
+     (i/simple-instruction
+       :int-add             ; The name of the instruction.
        [:int :int]          ; Takes two arguments from the :int stack
        :int                 ; Pushes a single result to the :int stack.
        0                    ; The instruction opens 0 code blocks.
        +)                   ; The function which turns arguments to results.
      {:puj.push.instruction/doc  "Pushes the sum of the top 2 ints onto the int stack."})
 
-   :int-sub
    (with-meta
-     (u/simple-instruction [:int :int] :int 0 -)
+     (i/simple-instruction :int-sub [:int :int] :int 0 -)
      {:puj.push.instruction/doc  "Pushes the difference of the top 2 ints onto the int stack."})
 
-   :int-mult
    (with-meta
```

Comment:
After about three of these, I would like to suggest we clean up the `with-meta` junk if possible. If we're going to be wrapping all these definitions in _other_ definitions (so they have docstrings), then we should consider a simpler function or macro that produces a documented function, not make every new instruction have this layered structure.

Reply:
I am not sure what you mean by layering, but I am fine with moving docstrings out of metadata.
We could keep a separate namespace with a map of instruction names to docstrings, but there is something to be said for having the docstring exist in a line that is nearby the instruction definition.

Given that docstrings are logically optional metadata, it seems like a natural implementation to me. We use qualified keywords that won't conflict with metadata from other libraries. Can you elaborate more on why you see this as a concern?

Reply:
I like this as metadata, unless @Vaguery has a good alternative.